packages/ui/src/common/components/Search/HighlightedText.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { memo, ReactElement } from 'react'
+import styled from 'styled-components'
+
+import { Colors } from '@/common/constants'
+import { isString } from '@/common/utils'
+
+type Node = ReactElement | string
+
+interface HighlightedTextProps {
+  pattern: RegExp | null
+  shorten?: boolean
+  children: string
+}
+export const HighlightedText = memo(({ pattern, shorten, children }: HighlightedTextProps) => {
+  const nodes = (pattern ? [...children.matchAll(pattern)] : []).reduceRight(
```

Comment:
About the `pattern` this is the RegExp used to find what to highlight. In this case it would only be a string with no special character but passing the RegExp rather than a string makes this component more reusable by allowing fancier RegExp. Also it allows to create the RegExp once and just pass it down by reference.